staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -267,6 +246,27 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope RequestSco
 			}
 		}
 
+		admit = admission.WithAudit(admit, ae)
+		if admit != nil && admit.Handles(admission.Delete) {
+			userInfo, _ := request.UserFrom(ctx)
+			attrs := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, "", scope.Resource, scope.Subresource, admission.Delete, len(options.DryRun) > 0, userInfo)
```

Comment:
This is being block is being moved down here so options will be created first
and ```isDryRunTrue(req.URL)``` can be replaced with ```len(options.DryRun) > 0```

Reply:
this will change some audit logging order, but I think it's ok.  The actual mutation comes later